core/src/test/scala/org/apache/spark/storage/LocalDirsSuite.scala

```diff
@@ -39,7 +39,7 @@ class LocalDirsSuite extends SparkFunSuite with BeforeAndAfter {
 
   private def assumeNonExistentAndNotCreatable(f: File): Unit = {
     try {
-      assume(!f.exists() && !f.mkdirs())
+      assert(!f.exists() && !f.mkdirs())
```

Comment:
@JoshRosen, this assume was intentional (see https://github.com/apache/spark/pull/17987/files#r116532209) because `/NENEXISTENT_PATH` can be a valid path. We should ideally fix the tests to make it invalid paths in all OSes.

Reply:
I am okay reverting this line for now too as it will be tricky to find out one invalid path in ["both Windows and UNIX-like systems (e.g. Linux, Mac OS)"](https://spark.apache.org/docs/latest/).

Reply:
I've reverted this line.